### PR TITLE
Add some tests ensuring views are not crashing

### PR DESCRIPTION
**Pull request by Elarnon**

### CODE REVIEW

**Comment by Elarnon** [comment]
I'm not sure. I like explicitly having all of the URLs in here because it allows to easily add additional tests (typically "is the context what we expect?"). I feel like this makes it easier to add regression tests.

File: mangaki/mangaki/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    def test_work_detail_redirect(self):
+        response = self.client.get('/work/{:d}'.format(self.anime.pk))
+        self.assertEqual(response.status_code, 301)  # 301 Moved Permanently
+        self.assertEqual(response.url, '/anime/{:d}'.format(self.anime.pk))
+
+    def test_work_detail_trailing_slash(self):
+        response = self.client.get('/anime/{:d}/'.format(self.anime.pk))
+        self.assertEqual(response.status_code, 404)  # 404 Not Found
+
+    def test_work_detail_nonexistant(self):
```
Comment (RaitoBezarius) [inline mangaki/mangaki/tests/test_views.py]: Missed typo… :D, it's `existent` I think.

**Comment by RaitoBezarius** [comment]
@Elarnon You have a point, I think that the only problem with this approach is it makes it harder to focus on the big picture, in a one test method, you are testing the content, the response code, the user model data at the same time.

Don't you think it'd make easier to do mistakes while writing tests? (assuming that the code is already not behaving properly).

Rather than having something like a list of URLs which should return 200, a list of URLs which should return 400, and so on.

Then, having more in-depth tests on the side-effects of these views? (you could argue that it'd take 2x time to run them, but we can always parallelize them on CircleCI and have fast tests / acceptance tests).

**Comment by Elarnon** [comment]
I'm a little bit concerned about speed, yes. But in general I think I disagree with your "big picture" objection. It simply is a different way of seeing things: are we testing Mangaki as a whole ("these URLs must return 200, these must return 400, these must only be available when logged") or are we testing views separately ("here is how this view should behave")?

I personally prefer the second approach, but ltimately I think that what matters the most is that things are tested, not the exact way with which the tests are written

**Comment by RaitoBezarius** [comment]
Right, my point is just : it makes it harder for someone to add basic tests like this view is supposed to _at least_ return 200 always.

The question is: do we want for every PRs to embed full view tests?

**Comment by Elarnon** [comment]
> The question is: do we want for every PRs to embed full view tests?

Yes? I know I am much more likely to add additional testing (e.g. what happens when the object does not exist, what happens when I'm not logged in, what happens when the event's date is in the past vs the future, etc. depending on the view) if I already have to write a small test checking that the view returns `200`. BTW returning 200 is not a given, e.g. it may require objects in the database or whatever, and I'd rather we spend time producing useful code for mangaki than architecturing a complex test architecture for URL return codes.

**Comment by RaitoBezarius** [comment]
That works for me, too. I prefer your approach only when we ask for contributors to provide their PRs with full view tests.

Let's write down in the CONTRIBUTING or track it into an issue, if it's not already done.

**Pull request merged**